### ISSUE TITLE
setutils.IndexedSet has symmetric difference()

### ISSUE THREAD

**Issue by spaghEddieDoyle** [mahmoud/boltons#252]
```python
>>> set('abc') - set('bcd')
{'a'}
>>> IndexedSet('abc') - IndexedSet('bcd')
IndexedSet(['a'])
>>> set('abc') - IndexedSet('bcd')
IndexedSet(['d'])
```
I would expect this to be due to `__sub__ = __rsub__ = difference` in the source code.

**Comment by mahmoud** [comment]
That's a good catch, and a good diagnosis. The wording had me a bit confused at first (bc symmetric_difference is also a thing). But fixed, will be released shortly, thank you :)

**Comment by spaghEddieDoyle** [comment]
Yes, I realize now that my wording is more than confusing -- it's just plain wrong. I was mostly trying to reference `__sub__ = __rsub__`. 

Thank you for your  help anyways!

**Comment by mahmoud** [comment]
Anytime!

**Comment by ankostis** [comment]
I just confirm that  41a926e  also fixes this faulty behavior that bit me:
```ipyhon
>>> set([1, 2]) - IndexedSet()
IndexedSet([])
```

The new behavior (unreleased as of 21 Jun 2020) is:
```
>>> set([1, 2]) - IndexedSet()
{1, 2}
```
...which seems ok to me.

+1 for the forthcoming release.
Any estimated date?

**Comment by mahmoud** [comment]
A fair question :) Just released boltons==20.2.0, thanks for the nudge!
